recipes-kernel/kernel-tests/kernel-performance-tests-files/run-cyclictest

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+source ./common.cfg
+
+function add_system_info()
+{
+    {
+	DESC=$(fw_printenv DeviceDesc)
+	DEVICE=${DESC#*=}
+	echo "# Device: $DEVICE"
+
+	echo -n "# Kernel: "
+	uname -a
+
+	echo -n "# Kernel parameters: "
+	cat /proc/cmdline
+
+	# save info on the current security mitigations
+	echo "# Security vulnerabilities settings: "
+	if [ -d "/sys/devices/system/cpu/vulnerabilities" ]; then
+	    for f in  /sys/devices/system/cpu/vulnerabilities/*; do
+		VULN=$(basename "$f")
+		if [ -f "$f" ]; then
+		    echo -n -e "#     $VULN:\t"
+		    cat "$f"
+		fi
+	    done
+	else
+	    echo "#     vulnerability information not published"
+	fi
+
+	# save info on the current C state settings
+	echo "# C-states settings: "
+	for CPU in /sys/devices/system/cpu/cpu[0-9]; do
+	    NCPU=$(basename "$CPU")
+	    echo "#     [$NCPU]"
+	    for CSTATE in $CPU/cpuidle/state[0-9]; do
+		NSTATE=$(basename "$CSTATE")
+		if [ -f "$CSTATE/name" ] && [ -f "$CSTATE/disable" ]; then
+                    NAME=$(cat "$CSTATE/name" 2>/dev/null)
+                    STATUS=$(cat "$CSTATE/disable" 2>/dev/null)
+
+                    echo -n "#     $NSTATE: $NAME: "
+                    if [ "$STATUS" = "0" ]; then
+                        echo "enabled"
+                    else
+                        echo "disabled"
+                    fi
+		else
+                    echo "#     information not available"
+		fi
+	    done
+	done
+    } >> "$1"
+}
+
+function run_cyclictest()
+{
+    LOG="results/cyclictest-$1-`date +'%Y_%m_%d-%H_%M_%S'`.log"
```

Comment:
Can we change this to an absolute path under `/tmp` or something? We haven't enforced a consistent working dir from which to run the ptests.

Reply:
I don't like using /tmp for this because it's volatile and limited in space, but I can look for a different place if desired.

IMHO having the results stored where the tests reside (/usr/lib/kernel-performance-tests/ptest/results/) is not the worst thing in the world. It makes them easier to retrieve since for these particular tests the log files are the important part not the pass/fail print.

Reply:
I think the test location would be fine. Another possibility might be to store them under $HOME for the running user. I was only objecting to using a blind, relative path - since who knows where the shell is when these tests are run.